### PR TITLE
[ceskatelevize] Live stream may not have title or description

### PR DESCRIPTION
**Pull request by sleep-walker**
This is the only thing which prevented run on live stream of Czech TV.

### CODE REVIEW

File: youtube_dl/extractor/ceskatelevize.py [hunk 1]
```diff
@@ -118,8 +119,12 @@ def _real_extract(self, url):
         req = sanitized_Request(compat_urllib_parse_unquote(playlist_url))
         req.add_header('Referer', url)
 
-        playlist_title = self._og_search_title(webpage)
-        playlist_description = self._og_search_description(webpage)
+        try:
+            playlist_title = self._og_search_title(webpage)
+            playlist_description = self._og_search_description(webpage)
```
Comment (dstftw) [inline youtube_dl/extractor/ceskatelevize.py]: Use `default=` instead. Also add some test.

**Comment by dstftw** [comment]
1. Live streams does not work properly - only current part of the stream is downloaded:
```
PS C:\dev\youtube-dl\master> py -2.7 .\youtube_dl\__main__.py http://www.ceskatelevize.cz/ivysilani/zive/ct4/
[CeskaTelevize] ct4: Downloading webpage
[CeskaTelevize] ct4: Downloading JSON metadata
[CeskaTelevize] ct4: Downloading JSON metadata
[CeskaTelevize] ct4: Downloading m3u8 information
[CeskaTelevize] ct4: Downloading m3u8 information
[download] Downloading playlist: ct4
[CeskaTelevize] playlist ct4: Collected 1 video ids (downloading 1 of them)
[download] Downloading video 1 of 1
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 3
[download] Destination: T Sport.mp4
[download] 100% of 10.77MiB in 00:09
[ffmpeg] Fixing malformated aac bitstream in "T Sport.mp4"
[download] Finished downloading playlist: ct4
```

2. Live streams should use `self._live_title` for title and set `is_live` in info dict.
3. `Živé vysílání ČT sport` and `Sledujete živé vysílání kanálu ČT sport.` can be used for title and description of playlist.

**Comment by sleep-walker** [comment]
Live streams does not work properly - only current part of the stream is downloaded.
Getting URL and passing to player works, which is my primary use. It would need some continuous read of playlist (it seems to change over the time). I tried to peek into other extractors but I haven't seen similar case (yet I know a very little about m3u8 or HLS). Any hint?

    Live streams should use self._live_title for title and set is_live in info dict.
Good point, fixed.

    Živé vysílání ČT sport and Sledujete živé vysílání kanálu ČT sport. can be used for title and description of playlist.
I don't think this is necessary, parsed `title` contains TV channel name and with `self._live_title` it gets timestamp so it is good as it is.

**Comment by dstftw** [comment]
>Any hint?

Just use `entry_protocol='m3u8'` in `self._extract_m3u8_formats` call for live streams to enable ffmpeg.

**Comment by sleep-walker** [comment]
> Just use entry_protocol='m3u8' in self._extract_m3u8_formats call for live streams to enable ffmpeg.

It helped, thanks.

File: youtube_dl/extractor/ceskatelevize.py [hunk 2]
```diff
@@ -77,6 +76,7 @@ class CeskaTelevizeIE(InfoExtractor):
     }]
 
     def _real_extract(self, url):
+        is_live = re.search('/ivysilani/zive/', url) != None
```
Comment (dstftw) [inline youtube_dl/extractor/ceskatelevize.py]: Live playlist items have `"type":"LIVE"` in JSON.

**Comment by dstftw** [comment]
Also squash commits when finished.

**Comment by sleep-walker** [comment]
Thanks for your guidance on this.

**Pull request closed without merging**